pkg/apis/topology/v1alpha2/types.go

```diff
@@ -0,0 +1,159 @@
+package v1alpha2
+
+import (
+	_ "github.com/gogo/protobuf/gogoproto"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TopologyManagerPolicy string
+
+const (
+	// LabelAlignmentScope represents the scope hardware topology alignment performed on the node.
+	LabelAlignmentScope = "topology.node.k8s.io/alignment-scope"
+)
+
+const (
+	// AlignmentScopeContainer means this node is aligning resources at container level
+	AlignmentScopeContainer = "container"
+	// AlignmentScopeContainer means this node is aligning resources at pod level
+	AlignmentScopePod = "pod"
+)
+
+const (
+	// LabelAlignmentConstraint represents how strictly resources are aligned when allocated on the node.
+	// Failing to align the resources using the constraint will cause workload admission failure.
+	LabelAlignmentConstraint = "topology.node.k8s.io/alignment-constraint"
+)
+
+const (
+	// AlignmentConstraintBesteffort signals the node will try to align resources, but no baseline is
+	// enforced and if no alignment is possible the workload will be admitted anyway.
+	AlignmentConstraintBestEffort = "besteffort"
+	// AlignmentConstraintMinimize signals the node will try to align resources minimizing the number
+	// of topology zones required.
+	AlignmentConstraintMinize = "minimize"
+	// AlignmentConstraintFull signals the node will only admit the workload if all the required resources
+	// can be allocated from a single topology zone
+	AlignmentConstraintFull = "full"
```

Comment:
Just a question if `full` means that we want to use only one zone, then the `AlignmentConstraintMinimize` shouldn't be called `AlignmentConstraintMaximize`?

Reply:
this is a very good point. I'm not fully happy with the naming here. To the best of my knowledge, the intent of the TM's `restricted` policy is to
```
[...] only allow allocations from the minimum number of NUMA nodes. Look at each resource request, see what the minimum number of NUMA nodes are required to satisfy that resource request. Allow alignment to that number of NUMA nodes for all resources. Fail otherwise.
```
IOW: the Topology Manager gather hints from all the providers. It determines the minimal size of NUMA nodes that can fullfill the request across all the providers. That becomes the target node size. If all the managers can allocate using that size, admit; fail otherwise.

"minimize" is meant to summarize this, but I must say it's still to obscure. I need more work in this area.

Reply:
Yeah both `minimize` and `maximize` will be confusing, maybe something like `Narrowest` like `TopologyManager` https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/topologymanager/policy.go#L154?

Reply:
Well, this is very good, thanks! Moving to "narrowest"